rpc/transport/ws/client.go

```diff
@@ -31,12 +31,17 @@ func NewWebSocketConnectionAsClient(url string) (*clientWebSocketConnection, err
 	return wsc, nil
 }
 
-func (wsc *clientWebSocketConnection) Request(id uint64, data []byte) ([]byte, error) {
+func (wsc *clientWebSocketConnection) Request(data []byte) ([]byte, error) {
 	// Any request payload type will do here since we are only interested in the id
 	responseChan := make(chan []byte, 1)
-	wsc.responseHandlers[id] = responseChan
+	unmarshaledRequest := serde.JsonRpcMessage{}
```

Comment:
It's a shame that the transport layer is coupled to the serde layer like this. I hadn't anticipated this problem when suggesting we remove `id` from the inputs to this method. 

I suppose the NATS transport didn't need to use the `id` since it has request/response "built in". Here, we're leaning heavily on the JSON-RPC `id` field to associate a reponse with a request. 

I don't want to block getting this work merged, but it would be nice to return to this issue and clean it up -- for example, not having the different role for `id` depending on transport / having a clear separation of concerns between the different layers in the rpc stack. 

Overall I'm not entirely sure how we should be organising this.

Reply:
Agreed that there is probably a better way to associate requests with responses. Filed https://github.com/statechannels/go-nitro/issues/1133 to keep track of this work.